core/src/main/java/org/hisp/dhis/android/core/trackedentity/internal/TrackedEntityAttributeReservedValueEndpointCallFactory.kt

```diff
@@ -0,0 +1,84 @@
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.android.core.trackedentity.internal
+
+import dagger.Reusable
+import org.hisp.dhis.android.core.arch.api.executors.internal.CoroutineAPICallExecutor
+import org.hisp.dhis.android.core.arch.call.factories.internal.QueryCoroutineCallFactoryImpl
+import org.hisp.dhis.android.core.arch.call.fetchers.internal.ListNoResourceWithErrorCatcherCallFetcher
+import org.hisp.dhis.android.core.arch.call.internal.GenericCallData
+import org.hisp.dhis.android.core.arch.call.processors.internal.CallProcessor
+import org.hisp.dhis.android.core.trackedentity.TrackedEntityAttributeReservedValue
+import javax.inject.Inject
+
+@Reusable
+internal class TrackedEntityAttributeReservedValueEndpointCallFactory @Inject internal constructor(
+    data: GenericCallData,
+    coroutineAPICallExecutor: CoroutineAPICallExecutor,
+    private val service: TrackedEntityAttributeReservedValueService,
+    private val handler: TrackedEntityAttributeReservedValueHandler,
+) : QueryCoroutineCallFactoryImpl<TrackedEntityAttributeReservedValue, TrackedEntityAttributeReservedValueQuery>(
+    data,
+    coroutineAPICallExecutor,
+) {
+
+    override suspend fun fetcher(query: TrackedEntityAttributeReservedValueQuery):
+        List<TrackedEntityAttributeReservedValue> {
+        val errorCatcher = TrackedEntityAttributeReservedValueCallErrorCatcher()
+
+        return object : ListNoResourceWithErrorCatcherCallFetcher<TrackedEntityAttributeReservedValue>(
+            coroutineAPICallExecutor = coroutineAPICallExecutor,
+            errorCatcher = errorCatcher,
+        ) {
+            override suspend fun call(): List<TrackedEntityAttributeReservedValue> {
+                return if (query.organisationUnit() == null) {
+                    service.generateAndReserve(
+                        query.trackedEntityAttributeUid(),
+                        query.numberToReserve(),
+                    )
+                } else {
+                    service.generateAndReserveWithOrgUnitCode(
+                        query.trackedEntityAttributeUid(),
+                        query.numberToReserve(),
+                        query.organisationUnit()!!.code()!!,
```

Comment:
The property `code()` might be null. We can check it in the "if" condition.
In java, the code won't fail with NullPointerException and the parameter will just be ignored because it is null. Now, it would throw a NullPointer